app/src/main/java/com/android/wardrobeManager/ui/images/filters/PrimarySecondary.java

```diff
@@ -1,31 +1,32 @@
 package com.android.wardrobeManager.ui.images.filters;
 
 import android.graphics.Bitmap;
-
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.PorterDuff;
+import android.graphics.PorterDuffXfermode;
 import com.android.wardrobeManager.ui.images.DesignFilterManager;
-
 import java.util.Map;
 
 public class PrimarySecondary implements DesignFilterManager.DesignFilter {
 
     protected Map<String, DesignFilterManager.DesignFilter> filters;
 
     public Bitmap filter(Bitmap bitmap, Bitmap ref, int[] colors) {
-        for(int n = 0; n < bitmap.getHeight(); n++) {
-            int color = 0;
-            int ymin = 0, ymax = bitmap.getHeight();
-            int div = bitmap.getHeight() / 3;
-            while (color + 1 < colors.length && ymin + div < n && n < ymax - div) {
-                ymin += div;
-                ymax -= div;
-                div /= 3;
-                color++;
-            }
-            for (int i = 0; i < bitmap.getWidth(); i++) {
-                if (ref.getPixel(i, n) != 0xFFFFFFFF) {
-                    bitmap.setPixel(i, n, colors[color]);
-                }
-            }
+        bitmap = ref.copy(ref.getConfig(), true);
```

Comment:
Same as in the stripes functions, this is done in design filter manager, so you don't have to do it here.